### DISCUSSION TITLE
Specify VSP RPM based on the scheduler?

### DISCUSSION THREAD

**Discussion by Tim-K5AC** [aqualinkd/AqualinkD#346] [General]
I noticed on your list of future releases it shows:
"Allow selecting of pre-defined VSP programs (Aqualink Touch & OneTouch protocols.)."

Would this allow someone to change the VSP programming based on a schedule?  I would like to run a weekly backwash program or to run the VSP at a lower RPM late at night for a few hours.  I'm wonder how this could be accomplished without having to program the pump directly?

Awesome product!!

Tim P

**Comment by sfeakes** [comment]
Yes It would.   At the moment you can already do this with AqualinkD scheduler, home automation hub or even a simple cron job.  Since you can already do it, it’s kind-a low on the priority of things to implement.

Jandy panel allows you to select a pre-defined RPM/GPM that has a name, and that’s what that item is.  Through the above methods you simply tell AqualinkD to set a specific RPM (rather than a name).

In AqualinkD web UI, click the time/date stamp at top right and the scheduler will come up, you can use that to schedule a VSP RPM/GPM.

**Reply by Tim-K5AC** [reply]
I can click on the scheduler, but I only see Filter Pump on/off, I don’t see a way to set the RPM from within the scheduler.  What am I missing?

**Reply by sfeakes** [reply]
If you have it configured as VSP, then you will get on/off/rpm (or GPM) under cmd.  If that's not there, then you don't have your pump configured correctly.
Do you see PRM in the status on the filter pump tile?  When you do a long click on the pump tile do you get option to change RPM?


![Screenshot 2024-09-09 at 7 37 51 PM](https://github.com/user-attachments/assets/01bc3dd9-ec56-49a3-b030-b5753a55d48b)

**Reply by Tim-K5AC** [reply]
Ok, I didn’t realize that when you clicked on command it brings up a dialog of ON, OFF,RPM.  I now have it working!

**Comment by mbarton98** [comment]
This was very helpful for me. I did not notice it in the wiki, but it helped me remove my reliance on the iAquaLink 2.0 software by figuring out how to set the pump speed (Pentair VSP).
